File: Service/Implement/DanhMucLayMauChiTieuService.cs
namespace Service.Implement
{
    public class DanhMucLayMauChiTieuService : BaseService<DanhMucLayMauChiTieu, IDanhMucLayMauChiTieuRepository>
    , IDanhMucLayMauChiTieuService
    {
        private readonly IDanhMucLayMauChiTieuRepository _DanhMucLayMauChiTieuRepository;
        public DanhMucLayMauChiTieuService(IDanhMucLayMauChiTieuRepository DanhMucLayMauChiTieuRepository) : base(DanhMucLayMauChiTieuRepository)
        {
            _DanhMucLayMauChiTieuRepository = DanhMucLayMauChiTieuRepository;
        }
        public override void Initialization(DanhMucLayMauChiTieu model)
        {
            BaseInitialization(model);
            if (model.ParentID == null)
            {
                model.ParentID = 1;
            }
        }
        public override DanhMucLayMauChiTieu Save(DanhMucLayMauChiTieu model)
        {
            DanhMucLayMauChiTieu DanhMucLayMauChiTieu = GetByName(model.Name);
            if (DanhMucLayMauChiTieu.ID > 0)
            {
                model = DanhMucLayMauChiTieu;
            }
            if (model.ID > 0)
            {
                Update(model);
            }
            else
            {
                Add(model);
            }
            return model;
        }
    }
}

File: CMS/src/app/shared/DanhMucLayMauChiTieu.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { DanhMucLayMauChiTieu } from 'src/app/shared/DanhMucLayMauChiTieu.model';
import { BaseService } from './Base.service';
@Injectable({
    providedIn: 'root'
})
export class DanhMucLayMauChiTieuService extends BaseService {

    List: DanhMucLayMauChiTieu[] | undefined;
    ListFilter: DanhMucLayMauChiTieu[] | undefined;
    FormData!: DanhMucLayMauChiTieu;

    DisplayColumns001: string[] = ['STT', 'ID', 'ParentID', 'Code', 'Name', 'Note', 'SortOrder', 'Active', 'Save'];

    constructor(public httpClient: HttpClient) {
        super(httpClient);
        this.Controller = "DanhMucLayMauChiTieu";
    }
    GetByParentIDAndEmptyToListAsync() {
        let url = this.APIURL + this.Controller + '/GetByParentIDAndEmptyToListAsync';
        if (this.BaseParameter.ParentID == 0) {
            return this.GetAllAndEmptyToListAsync();
        }
        const formUpload: FormData = new FormData();
        formUpload.append('data', JSON.stringify(this.BaseParameter));
        return this.httpClient.post(url, formUpload, { headers: this.Headers });
    }
}

